src/main/java/de/rabea/Game/Board.java

```diff
@@ -48,23 +82,118 @@ private int emptyCellsCount() {
         return count;
     }
 
-    public Cell switchMark(Cell mark) {
-        if (mark == Cell.X) {
-            return Cell.O;
-        } else {
-            return Cell.X;
+    public List<Integer> indexOfLastCellPerRow() {
+        int length = getDimension();
+        List<Integer> lastCellsPerRow = new ArrayList<Integer>();
+        for (int i = 1; i <= length; i ++) {
+            lastCellsPerRow.add(length * i - 1);
         }
+        return lastCellsPerRow;
     }
 
-    public List<Integer> emptyCells() {
-        List<Integer> emptyCells = new ArrayList<Integer>();
-        int i = 0;
-        for (Cell cell : cells) {
-            if (cell == EMPTY) {
-               emptyCells.add(i);
+    public int getDimension() {
+        double length = (double) cells().length;
+        double squareRoot = Math.sqrt(length);
+        return (int) squareRoot;
+    }
+
+    public int indexOfLastCell() {
+        return cells().length - 1;
+    }
+
+    public boolean isLastCell(int cell) {
+        return cell == indexOfLastCell();
+    }
+
+    public boolean isEndOfFirstOrSecondRow(int cell) {
+        for (int cellIndex : indexOfLastCellPerRow()) {
+            if (cell == cellIndex && !(isLastCell(cell))) {
+                return true;
             }
-            i++;
         }
-        return emptyCells;
+        return false;
+    }
+
+    private boolean anyLineMatchesAWinningCombination(Cell[] gameState, List<Integer> combo) {
+        return positionIsNotEmpty(gameState, combo) &&
+                sameMarkInFirstAndSecondPosition(gameState, combo) &&
+                sameMarkInSecondAndThirdPosition(gameState, combo);
+    }
+
+    private boolean positionIsNotEmpty(Cell[] gameState, List<Integer> combo) {
+        return gameState[combo.get(0)]!= Cell.EMPTY;
+    }
+
+    private boolean sameMarkInFirstAndSecondPosition(Cell[] gameState, List<Integer> combo) {
+        return gameState[combo.get(0)] == gameState[combo.get(1)];
+    }
+
+    private boolean sameMarkInSecondAndThirdPosition(Cell[] gameState, List<Integer> combo) {
+        return gameState[combo.get(1)] == gameState[combo.get(2)];
+    }
+
+    public List<List<Integer>> getAllLines() {
+        List<List<Integer>> allLines = new ArrayList<List<Integer>>();
+        for (List<Integer> row : getRows()) {
```

Comment:
may be nice to split these three for loops into separate methods called `addRows`, `addColumns`...?